app/scripts/modules/core/src/presentation/forms/Validation.ts

```diff
@@ -176,10 +205,14 @@ const createItemBuilder = (arrayBuilder: IValidationBuilder, index: number): IAr
 // Utility to provide a builder for array items. The provided iteratee will be invoked for every array item.
 const arrayForEach = (builder: (values: any) => IValidationBuilder, iteratee: ArrayItemValidator) => {
   return (array: any[], arrayLabel?: string) => {
+    // Silently ignore non-arrays (usually undefined). If strict type checking is desired, it should be done by a previous validator.
```

Comment:
I'm on the fence between:
- Baking too much checking into the validators vs
- Expecting an earlier validator (or a wrapped validator?) to have performed the validation?

Mostly answering for the `undefined` case where the field is not required, but if present then validate every array item.

Throwing on object is an interesting idea. I guess it would either be:
- `arrayForEach` throws or returns an error
- Expect an earlier validator to handle it: `.validate([isArray, arrayForEach])`

Something I haven't thought about yet is, what if a field can be one of multiple types (i.e. object or array, array or string, any, ¯\_(ツ)_/¯ )

Reply:
> what if a field can be one of multiple types (i.e. object or array, array or string, any, ¯_(ツ)_/¯ )

oh boy